src/coreclr/jit/liveness.cpp

```diff
@@ -2122,7 +2122,6 @@ void Compiler::fgComputeLifeLIR(VARSET_TP& life, BasicBlock* block, VARSET_VALAR
             case GT_STORE_BLK:
             case GT_STORE_DYN_BLK:
             case GT_JCMP:
-            case GT_CMP:
```

Comment:
This is what now allows us to remove `GT_CMP` nodes if they don't have `GTF_SET_FLAGS` set. Even before this PR we allowed that for the `GT_EQ` family of nodes, which were exactly the same as `GT_CMP` when void typed with `GTF_SET_FLAGS` set.